File: Microsoft.DotNet.Try.Protocol.Tests/StringExtensions.cs
// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Microsoft.DotNet.Try.Protocol.Tests
{
    public static class StringExtensions
    {
        public static string EnforceLF(this string source)
        {
            return source?.Replace("\r\n", "\n") ?? string.Empty;
        }
    }
}

File: src/microsoft-trydotnet-editor/src/stringExtensions.ts
// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

export function isNullOrUndefinedOrWhitespace(input: string): boolean {
    if (isNullOrUndefined(input)) {
        return true;
    }
    return input.replace(/\s/g, "").length < 1;
}

export function isNullOrUndefined(input: string): boolean {
    return input === undefined || input === null;
}
